libs/text-splitters/pyproject.toml

```diff
@@ -0,0 +1,103 @@
+[tool.poetry]
+name = "langchain-text-splitters"
+version = "0.0.1"
+description = "LangChain text splitting utilities"
+authors = []
+license = "MIT"
+readme = "README.md"
+repository = "https://github.com/langchain-ai/langchain"
+
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+langchain-core = "^0.1.28"
+
+[tool.poetry.group.lint]
+optional = true
+
+[tool.poetry.group.lint.dependencies]
+ruff = "^0.1.5"
+langchain-core = {path = "../core", develop = true}
+
+[tool.poetry.group.typing]
+optional = true
+
+[tool.poetry.group.typing.dependencies]
+mypy = "^1"
+lxml-stubs = "^0.5.1"
+types-requests = "^2.31.0.20240218"
+tiktoken = "^0.6.0"
+spacy = "^3.7.4"
+
+[tool.poetry.group.dev]
+optional = true
+
+[tool.poetry.group.dev.dependencies]
+jupyter = "^1.0.0"
+langchain-core = {path = "../core", develop = true}
+
+[tool.poetry.group.test]
+optional = true
+
+[tool.poetry.group.test.dependencies]
+# The only dependencies that should be added are
+# dependencies used for running tests (e.g., pytest, freezegun, response).
+# Any dependencies that do not meet that criteria will be removed.
+pytest = "^7.3.0"
+freezegun = "^1.2.2"
+pytest-mock  = "^3.10.0"
+pytest-watcher = "^0.3.4"
+pytest-asyncio = "^0.21.1"
+pytest-profiling = "^1.7.0"
+langchain-core = {path = "../core", develop = true}
+
+
+[tool.poetry.group.test_integration]
+optional = true
+dependencies = {}
+
+[tool.poetry.extras]
+extended_testing = ["jinja2"]
+
+[tool.ruff.lint]
+select = [
+  "E",  # pycodestyle
+  "F",  # pyflakes
+  "I",  # isort
+  "T201", # print
+]
+
+[tool.mypy]
+disallow_untyped_defs = "True"
+
+[[tool.mypy.overrides]]
```

Comment:
@efriis pretty nifty, lets you skip import errors for specific packages that don't have stubs